File: check.go
package main

// usage:
// go run main.go 1.2.3.4:8080

import (
	"fmt"
	"net"
	"os"
	"time"
)

func main() {
	target := "127.0.0.1:8080"
	if len(os.Args) >= 2 {
		target = os.Args[1]
	}

	c, err := net.DialTimeout("tcp", target, time.Second/2)
	if err != nil {
		fmt.Println(err)
		fmt.Println(target, "is closed")
	} else {
		c.Close()
		fmt.Println(target, "is open")
	}
}

File: check.rs
// build and run it
// rustc check.rs && ./check 127.0.0.2:8082

fn main() {
    let args: Vec<String> = std::env::args().collect();
    let mut target = "127.0.0.1:8080";
    if args.len() >= 2 {
        target = &args[1];
    }
    if !target.contains(":"){
        println!("bad target {}, missing port", target);
        return;
    }

    if let Ok(_stream) = std::net::TcpStream::connect(target) {
        println!("{} is open", target);
    } else {
        println!("{} is closed", target);
    }
}
